src/storage/src/upsert_continual_feedback.rs

```diff
@@ -432,23 +433,18 @@ where
             // loop. More of our stash can become eligible for draining both
             // when the source-input frontier advances or when the persist
             // frontier advances.
+            if !stash.is_empty() {
+                let cap = stash_cap.as_mut().expect("missing capability for non-empty stash");
```

Comment:
AFAIU this change here seems to be the fix for the original issue? Previously because `stash` was a hashmap we could end up with the same key with two different values spread across two different `(cap, updates)` sets. Thus `drain_staged_input` could get called multiple times with the same `persist_upper` and process the same key.

Now because we process the entire `stash` at once, and we have the check on line 423, we're guaranteed to only call `drain_staged_input` with a `DrainStyle::ToUpper` with a given `persist_upper`, a single time.

If that understanding is correct, do you mind adding a comment here explaining how this is load bearing? Or maybe how these pieces work together to maintain the new correctness property documented on `drain_staged_input`?

Reply:
Technically having the `HashMap` would be fine too if the managed vectors of updates were appropriately distributed by timestamp. So even though this change fixes the issue it's not load bearing in the sense that it must be this way.

To avoid going another CI round I'll merge as is and we will revisit this code soon since we want to simplify its logic further and this section will definitely be affected.